### PR TITLE
Fixes bug when loading user-provided msg-template.

### LINKED ISSUE #5636: Using custom braces in message template does not work
### Bug description

Have any list of errors:

On pylint 1.7 w/ python3.6 - I am able to use this as my message template
```
$ pylint test.py --msg-template='{{ "Category": "{category}" }}'
No config file found, using default configuration
************* Module [redacted].test
{ "Category": "convention" }
{ "Category": "error" }
{ "Category": "error" }
{ "Category": "convention" }
{ "Category": "convention" }
{ "Category": "convention" }
{ "Category": "error" }
```

However, on Python3.9 with Pylint 2.12.2, I get the following:
```
$ pylint test.py --msg-template='{{ "Category": "{category}" }}'
[redacted]/site-packages/pylint/reporters/text.py:206: UserWarning: Don't recognize the argument '{ "Category"' in the --msg-template. Are you sure it is supported on the current version of pylint?
  warnings.warn(
************* Module [redacted].test
" }
" }
" }
" }
" }
" }
```

Is this intentional or a bug?

### Configuration

_No response_

### Command used

```shell
pylint test.py --msg-template='{{ "Category": "{category}" }}'
```

### Pylint output

```shell
[redacted]/site-packages/pylint/reporters/text.py:206: UserWarning: Don't recognize the argument '{ "Category"' in the --msg-template. Are you sure it is supported on the current version of pylint?
  warnings.warn(
************* Module [redacted].test
" }
" }
" }
" }
" }
" }
```

### Expected behavior

Expect the dictionary to print out with `"Category"` as the key.

### Pylint version

```shell
Affected Version:
pylint 2.12.2
astroid 2.9.2
Python 3.9.9+ (heads/3.9-dirty:a2295a4, Dec 21 2021, 22:32:52) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]


Previously working version:
No config file found, using default configuration
pylint 1.7.4, 
astroid 1.6.6
Python 3.6.8 (default, Nov 16 2020, 16:55:22) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by clavedeluna**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

Closes #5636

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Nice !

**Comment by DanielNoord** [comment]
@clavedeluna Could you make the news message a little more descriptive? This doesn't really help anybody looking at the changelog to see what has changed. Fix itself LGTM!

File: doc/whatsnew/fragments/5636.bugfix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fixes bug when loading user-provided msg-template.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/5636.bugfix:1]:
```suggestion
Using custom braces in ``msg-template`` will now work properly 
```

**Review by Pierre-Sassoulas — DISMISSED** [review]

**Comment by DanielNoord** [comment]
@clavedeluna Could you fix the pre-commit failure?

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**